Gems/Terrain/Code/Source/TerrainRenderer/ClipmapBounds.h

```diff
@@ -0,0 +1,140 @@
+{
+
+    struct ClipmapBoundsDescritor
+    {
+        //! Size of the clipmap in width / height.
+        uint32_t m_size = 1024;
+
+        //! Current center location of the clipmap
+        AZ::Vector2 m_center = AZ::Vector2::CreateZero();
+
+        //! Unused area around edges of the clipmap. Center point must be moved by
+        //! at least this much before an update is triggered. This allows for larger
+        //! but less frequent updates.
+        uint32_t m_margin = 4;
+
+        //! Scale of the clip map compared to the world. A scale of 0.5 means that
+        //! a clipmap of size 1024 would cover 512 meters.
+        float m_scale = 1.0f;
+    };
+
+    struct ClipmapBoundsRegion
+    {
+        //! The world bounds of the updated region. Z is ignored.
+        AZ::Aabb m_worldAabb;
+
+        //! The clipmaps bounds of the updated region. Will always be between 0 and size.
+        //! Min inclusive, max exclusive. i.e., a single pixel would be min [0, 0], max [1, 1].
+        Aabb2i m_localAabb;
+
+        bool operator==(const ClipmapBoundsRegion& other) const;
+        bool operator!=(const ClipmapBoundsRegion& other) const;
+    };
+
+    // This class manages a single clipmap region. This is typically a texture which contains data
+    // around a point like the camera. The underlying texture never actually moves, and since it
+    // wraps it can be thought of like a repeating grid. This makes looking up data in the clipmap
+    // trivial since it's just the world coordinate scaled by some amount, however only data inside
+    // the clipmap bounds will actually be valid. This technique also allows for only the edge areas
+    // of the clipmap to be updated as the center point moves around the world.
+    //
+    // The edges of the clipmap region will typically run through the texture, dividing it into 4
+    // regions, except in cases where the clipmap region happens to be aligned with the underlying
+    // grid. This means whenever some bounding box needs to be updated in the clipmap, it may actually
+    // translate to 4 different areas of the underlying texture - one for each quadrant.
+    //
+    // This class aids in figuring out which areas of a clipmap need to be updated as its center point
+    // moves around in the world, and can map a single region that needs to be updated into several
+    // separate regions for each quadrant.
+    // _____________________________
+    // |      |      |      |      |      Clipmap
+    // |      |      |      |      |      Texture
+    // |______|______|______|______|      ______ 
+    // |      |     _|____  |      |     | |    |
+    // |      |    | |    | |      |     |_|____|
+    // |______|____|_|_*__|_|______|     |_|____|
+    // |      |    |_|____| |      |
+    // |      |      |      |      |
+    // |______|______|______|______|
+    // |      |      |      |      |
+    // |      |      |      |      |
+    // |______|______|______|______|
```

Comment:
Neato. But one suggestion based on a few warnings i've observed with gcc. It doesnt affect this particular ascii art, but if you use the c++ line '//' comment token, and your ascii art happens to end with a backslash, it will warn you (legitimately) that the backslash will be interpreted as a continuation to the next line. So for areas that have nice ascii art for descriptions, i'd suggest using the `/*  */` block style commenting instead. It will also clean up the left side of the image.

```suggestion
    /*
       This class aids in figuring out which areas of a clipmap need to be updated as its center point
       moves around in the world, and can map a single region that needs to be updated into several
       separate regions for each quadrant.
       _____________________________
       |      |      |      |      |      Clipmap
       |      |      |      |      |      Texture
       |______|______|______|______|      ______ 
       |      |     _|____  |      |     | |    |
       |      |    | |    | |      |     |_|____|
       |______|____|_|_*__|_|______|     |_|____|
       |      |    |_|____| |      |
       |      |      |      |      |
       |______|______|______|______|
       |      |      |      |      |
       |      |      |      |      |
       |______|______|______|______|
    */
```

Reply:
Good idea - I'll update.